app/controllers/registrations_controller.rb

```diff
@@ -7,4 +7,15 @@ def new
       redirect_to root_path, alert: 'Sign up is only allowed via invitation.'
     end
   end
+
+  def update
+    if params[:student][:email] != current_student.email && current_student.valid_password?(params[:student][:current_password])
+      begin
+        current_student.update_close_email(params[:student][:email])
+      rescue Exception => e
```

Comment:
Don't ever rescue all exceptions, because then if you have an unexpected exception, your program will do crazy things and you will have no idea why. Instead, rescue exceptions of the exception class you expect to see raised and handle it appropriately.